src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -301,6 +317,39 @@ private async Task PreRunConditions()
             }
         }
 
+        private bool HostJsonExist()
+        {
+            var hostJsonPath = Path.Combine(Environment.CurrentDirectory, "host.json");
+            return File.Exists(hostJsonPath);
+        }
+
+        private bool BundleConfigurationExists()
+        {
+            var hostJsonPath = Path.Combine(Environment.CurrentDirectory, "host.json");
+            var hostJson = FileSystemHelpers.ReadAllTextFromFile(hostJsonPath);
+            return hostJson.Contains("extensionBundle", StringComparison.OrdinalIgnoreCase);
+        }
+
+        private bool IsPreCompiledFunctionApp()
+        {
+            bool isPrecompiled = false;
+            foreach (var directory in FileSystemHelpers.GetDirectories(Environment.CurrentDirectory))
```

Comment:
should this ScriptRootPath instead of current directory?

Reply:
The current directory is actually managed by core tools and changes based on the type of app. For example for dotnet it changes to `/bin/output/bin`. Changing to the scriptroot would not work or would require larger refactoring.